File: backend/app/schemas/expense.py
"""Pydantic schemas for expenses."""

from datetime import date
from pydantic import BaseModel, Field


class ExpenseCreate(BaseModel):
    """Create a new expense."""
    amount: float = Field(..., gt=0, description="Expense amount")
    description: str | None = Field(None, max_length=500)
    category_id: int | None = None
    expense_date: date = Field(default_factory=date.today)
    receipt_url: str | None = None


class ExpenseUpdate(BaseModel):
    """Update an existing expense."""
    amount: float | None = Field(None, gt=0)
    description: str | None = None
    category_id: int | None = None
    expense_date: date | None = None


class ExpenseResponse(BaseModel):
    """Expense response with category info."""
    id: int
    amount: float
    currency: str
    description: str | None
    expense_date: date
    category_id: int | None = None
    category_name: str | None = None
    category_icon: str | None = None
    source: str
    created_at: str

    model_config = {"from_attributes": True}


class ExpenseSummary(BaseModel):
    """Aggregated expense summary."""
    total: float
    count: int
    by_category: dict[str, float] = {}
    period: str  # "daily", "weekly", "monthly"
    start_date: str
    end_date: str


class CategoryResponse(BaseModel):
    """Category response."""
    id: int
    name: str
    icon: str | None
    color: str | None
    is_custom: bool

File: frontend/src/types/expense.ts
// Expense-related types

export interface Category {
  id: number
  name: string
  icon: string | null
  color: string | null
  is_custom: boolean
}

export interface Expense {
  id: number
  amount: number
  currency: string
  description: string | null
  expense_date: string
  category_id: number | null
  category_name: string | null
  category_icon: string | null
  source: 'manual' | 'whatsapp' | 'bank_sync'
  created_at: string
}

export interface ExpenseSummary {
  total: number
  count: number
  by_category: Record<string, number>
  period: 'daily' | 'weekly' | 'monthly'
  start_date: string
  end_date: string
}

export interface ExpenseCreate {
  amount: number
  description?: string
  category_id?: number
  expense_date?: string
}

export interface ExpenseUpdate {
  amount?: number
  description?: string
  category_id?: number
  expense_date?: string
}

// AI-Powered Features
export interface AIStatus {
  configured: boolean
  model: string
  features: string[]
}

export interface ParsedExpense {
  amount: number | null
  description: string | null
  category: string
  category_id: number | null
  confidence: number
}

export interface ReceiptData {
  merchant: string | null
  amount: number | null
  items: string[]
  date: string | null
  category: string
  category_id: number | null
  confidence: number
}

export interface SmartExpenseCreate {
  text?: string
  amount?: number
  description?: string
  category_id?: number
  expense_date?: string
  use_ai?: boolean
}
